src/ArcCommander/APIs/ArcAPI.fs

```diff
@@ -118,79 +120,82 @@ module ArcAPI =
+
+                match studies with
+                | Some studies ->
+                    match studies |> Seq.tryFind (API.Study.getAssays >> Option.defaultValue [] >> API.Assay.existsByFileName assayFileName) with
+                    | Some study -> 
+                        study
+                        |> API.Study.mapAssays (API.Assay.updateByFileName API.Update.UpdateByExistingAppendLists assay)
+                        |> API.Study.mapFactors (List.append factors >> List.distinctBy (fun f -> f.Name))
+                        |> API.Study.mapProtocols (List.append protocols >> List.distinctBy (fun p -> p.Name))
+                        |> API.Study.mapContacts (List.append persons >> List.distinctBy (fun p -> p.FirstName,p.LastName))
+                        |> fun s -> API.Study.updateBy ((=) study) API.Update.UpdateAll s studies
+                    | None ->
+                        Study.fromParts (Study.StudyInfo.create a "" "" "" "" "" []) [] [] factors [assay] protocols persons
+                        |> API.Study.add studies
+                | None ->                   
+                    [Study.fromParts (Study.StudyInfo.create a "" "" "" "" "" []) [] [] factors [assay] protocols persons]
+                |> API.Investigation.setStudies investigation
+                |> updateInvestigationAssays t
+            | [] -> investigation
                 
-            let output = updateInvestigationAssays (assayNames |> List.ofArray) investigation
+        let output = updateInvestigationAssays (assayNames |> List.ofArray) investigation
 
-            if containsFlag "ProcessSequence" arcArgs then
+        if containsFlag "ProcessSequence" arcArgs then
 
-                let output = 
-                    output.Studies 
-                    |> Option.defaultValue [] |> List.collect (fun s -> 
-                        s.Assays
-                        |> Option.defaultValue [] |> List.collect (fun a -> 
-                            a.ProcessSequence |> Option.defaultValue []
-                        )
+            let output = 
+                output.Studies 
+                |> Option.defaultValue [] |> List.collect (fun s -> 
+                    s.Assays
+                    |> Option.defaultValue [] |> List.collect (fun a -> 
+                        a.ProcessSequence |> Option.defaultValue []
                     )
+                )
 
-                match tryGetFieldValueByName "Path" arcArgs with
-                | Some p -> ArgumentProcessing.serializeToFile p output
-                | None -> ()
+            match tryGetFieldValueByName "Path" arcArgs with
+            | Some p -> ArgumentProcessing.serializeToFile p output
+            | None -> ()
 
-                System.Console.Write(ArgumentProcessing.serializeToString output)
-            else 
+            //System.Console.Write(ArgumentProcessing.serializeToString output)
+            log.Debug(ArgumentProcessing.serializeToString output)
```

Comment:
Previously it was possible to completely silence the ArcCommander and still get the output from above printed into the console. Is this still possible?

Reply:
Yes.